lib/vagrant-service-manager/installer.rb

```diff
@@ -0,0 +1,28 @@
+require_relative 'adb_executor'
+require_relative 'cdk_executor'
+
+module VagrantPlugins
+  module ServiceManager
+    module Executor
+      def self.build(type, machine, ui, options)
```

Comment:
All this logic is testable without going all the way with the starting a new VM. I think we need to introduce unit tests now. Here is a commit - https://github.com/hferentschik/adb-vagrant-registration/commit/18679a122c11c03072ee63617c48e67f2e97bb39 - which introduces minitest for vagrant-registration. It is a very simple setup and builds on mocking out some of the core classes of Vagrant. This way you can execute commands, capabilities, etc without going end to end. For example a test could make sure that the right executor is returned. In test you basically record which response the os_variant is supposed to return and then you can assert what Executor is returned. You can also test what happens when the variant is not at all adb or cdk. Just as an example.